components/producers/github-dependabot/main.go

```diff
@@ -0,0 +1,169 @@
+}
+
+func main() {
+	flag.StringVar(&cfg.RepositoryOwner, "repository-owner", "", "The owner of the GitHub repository")
+	flag.StringVar(&cfg.RepositoryName, "repository-name", "", "The name of the GitHub repository")
+	flag.StringVar(&cfg.GitHubToken, "github-token", "", "The GitHub token used to authenticate with the API")
+	flag.StringVar(&cfg.Ref, "reference", "", "The Ref/branch to get alerts for")
+	flag.StringVar(&cfg.Severity, "severity", "", "If specified, only code scanning alerts with this severity will be returned. Possible values are: critical, high, medium, low, warning, note, error")
+	flag.StringVar(&cfg.Ecosystem, "ecosystem", "", "If specified, a comma separated list of at least one of composer, go, maven, npm, nuget, pip, pub, rubygems, rust")
+	if err := producers.ParseFlags(); err != nil {
+		log.Fatal(err)
+	}
+
+	apiClient := wrapper.NewClient(cfg.GitHubToken)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	alerts, err := listAlertsForRepo(ctx, apiClient, cfg.RepositoryOwner, cfg.RepositoryName, cfg.Severity, cfg.Ecosystem)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	issues := parseIssues(alerts)
+
+	if err := producers.WriteDraconOut("github-code-scanning", issues); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func listAlertsForRepo(ctx context.Context, apiClient wrapper.Wrapper, owner, repo, severity, ecosystem string) ([]*github.DependabotAlert, error) {
+	open := "open"
+	opt := &github.ListAlertsOptions{
+		State:     &open,
+		Severity:  &severity,
+		Ecosystem: &ecosystem,
+		ListOptions: github.ListOptions{
+			PerPage: 30,
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	defer cancel()
+
+	var allAlerts []*github.DependabotAlert
+	for {
+		alerts, resp, err := apiClient.ListRepoDependabotAlerts(ctx, owner, repo, opt)
+		if err != nil {
+			return nil, err
+		}
+
+		allAlerts = append(allAlerts, alerts...)
+
+		if resp.NextPage == 0 {
+			break
+		}
+
+		opt.ListOptions.Page = resp.NextPage
+	}
+	slog.Info("Successfully fetched alerts", slog.Int("count", len(allAlerts)), slog.String("repository", owner+"/"+repo))
+	return allAlerts, nil
+}
+
+func parseIssues(alerts []*github.DependabotAlert) []*v1proto.Issue {
+	issues := make([]*v1proto.Issue, 0, len(alerts))
+	for _, alert := range alerts {
+		ecosystem := *(alert.GetSecurityVulnerability().Package.Ecosystem)
+		if ecosystem == "pip" {
+			ecosystem = "pypi"
+		}
+
+		cwe := []int32{}
+		for _, c := range alert.SecurityAdvisory.CWEs {
+			numberOnly := strings.ReplaceAll(*c.CWEID, "CWE-", "")
+			cweNum, err := strconv.Atoi(numberOnly)
+			if err != nil {
+				slog.Error("could not extract cwe number from ", slog.String("cweID", *c.CWEID))
+				continue
+			}
+
+			cwe = append(cwe, int32(cweNum))
+		}
+		purl := producers.GetPURLTarget(ecosystem, "", *alert.GetSecurityVulnerability().Package.Name, "", packageurl.Qualifiers{}, "")
+		slog.Info("Handling alert for", slog.String("purl", purl))
+		cve := ""
```

Comment:
super ⛏️  but we could use a `var ()` block to define all these vars inline

Would look slicker imo

```
var (
    cve string
    summary string 
    ...
)
```